src/PSRule.Rules.Azure/rules/Azure.AppGw.Rule.ps1

```diff
@@ -77,12 +77,8 @@ Rule 'Azure.AppGw.UseHTTPS' -Type 'Microsoft.Network/applicationGateways' -Tag @
 }
 
 # Synopsis: Application gateways deployed with V2 SKU(Standard_v2, WAF_v2) should use availability zones in supported regions for high availability.
-Rule 'Azure.AppGw.AvailabilityZone' -Type 'Microsoft.Network/applicationGateways' -Tag @{ release = 'GA'; ruleSet = '2021_09'; } {
-
-    # Availability zones only available in v2 tiers of Application Gateway
-    $isV2Sku = $Assert.In($TargetObject, 'Properties.sku.tier', @('Standard_v2', 'WAF_v2')).Result;
-
-    if (-not $isV2Sku -or [string]::IsNullOrEmpty($TargetObject.Location)) {
+Rule 'Azure.AppGw.AvailabilityZone' -Type 'Microsoft.Network/applicationGateways' -If { IsAppGwV2Sku } -Tag @{ release = 'GA'; ruleSet = '2021_09'; } {
+    if ([string]::IsNullOrEmpty($TargetObject.Location)) {
```

Comment:
Not sure if I need this check anymore. To me it might be more useful checking a valid location is passed inside the precondition as well.

Reply:
Maybe something like checking a valid location would be more appropriate?

Reply:
Might be better to just check location against `providers.json`.

Reply:
Can probably just add `[AllowEmptyString()]` to `GetNormalLocation` and remove this check I'm doing in the AZ rules.